### ISSUE TITLE
Add model support

### ISSUE THREAD

**Issue by Jcarminio** [dfinke/PSAI#49]
Please add GPT-4o-Mini to the list of options. It offers the most cost-effective solution with a 128K token capacity

**Comment by Jcarminio** [comment]
Ohh thank you for answering!

I was using New-OpenAIChat + New-Agent + Get-AgentResponse

```
$chat = New-OpenAIChat -model "gpt-3.5-turbo-16k"
$webTool = New-TavilyAITool
$agent = New-Agent -Tools $webTool -LLM $chat -ShowToolCalls
$agent | Get-AgentResponse $Question
```
I've just started using it today so I did not know this!

**Comment by dfinke** [comment]
Cool! I am prepping to merge a provider/model plugin architecture. This enables working with Ollama, Gemini etc. You can even add your own as new ones come online.

**Comment by Jcarminio** [comment]
That would be awesome! I am an infrastructure administrator so gemini's large tokens capacity is really useful!

**Comment by dfinke** [comment]
Yes! plus, you can test/eval different models with the same agent (diff prompts) in one script, one of many scenarios.

**Comment by Jcarminio** [comment]
Nice job! This will help a lot of us.

**Comment by dfinke** [comment]
Could you say a bit more on why that is useful?

As a dev, I like options. Using OpenAI and Azure OAI serves my needs tho, till the other models really peel away from the pack.

**Comment by Jcarminio** [comment]
I think having a centralized tool such as this model, where one can use different AI models for several use cases (in my case, 4o mini is good for code writing and gemini for identifying specific things in large files / outputs) is good for fast problem solving, where I spend less time trying to find the right commands / errors.
I work with SCOM / SCCM / SharePoint / Exchange / Active Directory which tend to have lot's of errors daily. PowerShell is one of the languages that is most integrated in all of these Services so having AI directly in it saves time.

**Comment by dfinke** [comment]
awesome summary! thank you.

Yes - wielding AI integration at the CLI is the quickest way to test, integrate, etc. 
One of the reasons I built it 🙂

Having access at your fingertips to other providers/models does help you pivot rapidly.

Ask question and raise issues

Thank you for your response.

**Comment by dfinke** [comment]
would love to hear how you apply this in those enviroments and how it goes.

**Comment by Jcarminio** [comment]
I'm still experimenting with it. Today, I also gained the responsibility of becoming the Spectrum Protection Administrator (backup for my company), and I have to check if all the backups were successful and find any anomalies. I designed a PowerShell script that uses the TSM API to get all the backup data, so I’m thinking of using PSAI to find those anomalies and trigger some sort of action to let me know. I still haven't had the time to put it all together.
